thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/HashService.js

```diff
@@ -110,12 +110,41 @@ HashService.prototype = {
     console.log('routeTo' + controllerName);
     this.currentControllerName = controllerName;
     this.currentController = this.controllerNameToControllerMap[controllerName];
-    this.refreshWindowHashForRouting(controllerName);
-  },
-  onHashChangeEventHandler : function() {
-    console.log('OnHashChangeEventhandler');
-    console.log(this.params);
-    console.log(this.currentController);
     this.currentController.handleAppEvent();
+  },
+
+  /**
+   * determine based on the hash change event if a transition is needed
+   * @param  {string} options.newURL [newURL of hash change event]
+   * @return {boolean} - returns true if a transition is needed]
+   */
+  shouldTransition({ newURL }) {
+    const [, transitionTo, paramsTo] = newURL.split(/(?:[#?]|[&]*rand)/);
```

Comment:
can you add a simple string that this is trying to split. I possible avoid complex regex. It's ok to break it up into multiple steps.